src/test/java/com/sproutsocial/nsq/RoundRobinDockerTestIT.java

```diff
@@ -0,0 +1,143 @@
+package com.sproutsocial.nsq;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class RoundRobinDockerTestIT extends BaseDockerTestIT {
+    private static final Logger LOGGER = getLogger(RoundRobinDockerTestIT.class);
+    private Subscriber subscriber;
+    private Publisher publisher;
+    private TestMessageHandler handler;
+
+    @Override
+    public void setup() {
+        super.setup();
+        Util.sleepQuietly(500);
+        //This needs to be crazy long because it can take up to 1 min for nsq-lookup to remove the bad nodes and add them back again so we can reconnect
```

Comment:
Hrm, I wonder if we can configure this in the `nsqlookupd` startup parameters.

Reply:
Sorry, this was a red herring that made it in as a comment.  The issue was just the connections themselves hanging around too long due to a very long socket timeout.  Will correct.  
This timeout is currently effectively hard coded, and I am hesitant to change it.